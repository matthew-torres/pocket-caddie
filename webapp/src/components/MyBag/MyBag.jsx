import React from 'react';
import { DataGrid } from '@mui/x-data-grid';
import TableContainer from '@mui/material/TableContainer';
import Table from '@mui/material/Table';
import Paper from '@mui/material/Paper';
import TableHead from '@mui/material/TableHead';
import TableRow from '@mui/material/TableRow';
import TableCell from '@mui/material/TableCell';
import TableBody from '@mui/material/TableBody';
import Checkbox from '@mui/material/Checkbox';
import useRowsData from '../../retrieveBagData';
import { Box, IconButton,Typography } from '@mui/material';
import { useState } from 'react';
import { ThemeProvider, useTheme } from '@mui/material/styles';
import DeleteIcon from '@mui/icons-material/Delete';
import AddIcon from '@mui/icons-material/Add';
import CreateClub from "../CreateClub/CreateClub";
import Dialog from '@mui/material/Dialog';
import { globalTheme } from '../../theme';


const columns = [

    { field: 'type', headerName: 'Type', width: 130},
    { field: 'brand', headerName: 'Brand', width: 13},
    { field: 'name', headerName: 'Name', width: 130},
  ];

export default function MyBag() {

    const url = process.env.VITE_API_URL;
    const theme = useTheme();
    let rows = useRowsData();
    const [open, setOpen] = useState(false);

    const handleAdd = () => {
        setOpen(true);
      };
    
      const handleClose = () => {
        setOpen(false);
      };

      const handleDelete = (event) => {
        if (selectionModel.length === 1) {
          event.preventDefault();
          axios.delete(url+'api/user/club/'+selectionModel[0], {headers: {'Content-Type': 'application/json',Authorization: `Bearer ${sessionStorage.getItem('token')}`}})
          .then(response => { 
              console.log(response)
              window.location.reload();
          })
          .catch(error => {
              console.log(error)
          })
      } else {
        axios.delete(url+'api/user/clubs', {data: { selectionModel: selectionModel },headers: {'Content-Type': 'application/json', Authorization: `Bearer ${sessionStorage.getItem('token')}`}})
          .then(response => { 
              console.log(response)
              window.location.reload();
          })
          .catch(error => {
              console.log(error)
          })
      };};
  
    
    return (
    <ThemeProvider theme={globalTheme}>
        <React.Fragment>
            <Typography variant="h8" color={theme.palette.text.primary} fontWeight="bold" sx={{justifyContent: "left", m: "0 0 10px 0" }}>
                My Bag 
            </Typography>
            <TableContainer component={Paper} sx={{ backgroundColor: theme.palette.background.default }}>
                <Table stickyHeader>
                    <TableHead>
                    <TableRow>
                        {columns.map((column) => (
                        <TableCell key={column.field} sx={{ fontWeight: 'bold' }}>{column.headerName}</TableCell>
                        ))}
                    </TableRow>
                    </TableHead>
                    <TableBody>
                    {rows.map((row) => (
                        <TableRow key={row.clid}>
                        {columns.map((column) => (
                            <TableCell key={column.field} sx={{ color: theme.palette.text.secondary }}>
                            {row[column.headerName]}
                            </TableCell>
                        ))}
                        </TableRow>
                    ))}
                    </TableBody>
                </Table>
                <Dialog open={open} onClose={handleClose} PaperProps={{sx: {backgroundColor: theme.palette.background.default }}}>
                    <CreateClub/>
                </Dialog>
            </TableContainer>
                <Box display="flex" justifyContent="flex-start">
                    <IconButton type="submit" variant='outlined' onClick={handleDelete} color="primary"><DeleteIcon/></IconButton>
                    <IconButton type="submit" variant='outlined' onClick={handleAdd} color="primary"><AddIcon/></IconButton>
                </Box>
        </React.Fragment>
    </ThemeProvider>
      );
}