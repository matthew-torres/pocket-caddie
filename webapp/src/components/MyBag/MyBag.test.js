import React from "react";
import renderer from 'react-test-renderer';
import { Renderer } from "@testing-library/react-hooks";
import MyBag from '../MyBag/MyBag';
import MockAdapter from 'axios-mock-adapter';
import instance from "../../axios_instance";
import { render, waitFor } from "@testing-library/react";


jest.mock('../../retrieveBagData', () => {
    return jest.fn(() => [
      { clid: 1, type: 'Type1', brand: 'Brand1', name: 'Name1' },
      { clid: 2, type: 'Type2', brand: 'Brand2', name: 'Name2' }
    ]);
  });
  
  describe('MyBag component', () => {
    it('renders MyBag correctly', async () => {

    const component = renderer.create(<MyBag />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
    });
  });