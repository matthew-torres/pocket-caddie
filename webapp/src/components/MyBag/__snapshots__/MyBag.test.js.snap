// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MyBag component renders MyBag correctly 1`] = `
[
  <span
    className="MuiTypography-root MuiTypography-h8 css-z8mfk0-MuiTypography-root"
  >
    My Bag
  </span>,
  <div
    className="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiTableContainer-root css-qdfd52-MuiPaper-root-MuiTableContainer-root"
  >
    <table
      className="MuiTable-root MuiTable-stickyHeader css-3ay9qw-MuiTable-root"
      role={null}
    >
      <thead
        className="MuiTableHead-root css-15wwp11-MuiTableHead-root"
        role={null}
      >
        <tr
          className="MuiTableRow-root MuiTableRow-head css-115oldd-MuiTableRow-root"
          role={null}
        >
          <th
            aria-sort={null}
            className="MuiTableCell-root MuiTableCell-head MuiTableCell-stickyHeader MuiTableCell-sizeMedium css-5xj66b-MuiTableCell-root"
            scope="col"
          >
            Type
          </th>
          <th
            aria-sort={null}
            className="MuiTableCell-root MuiTableCell-head MuiTableCell-stickyHeader MuiTableCell-sizeMedium css-5xj66b-MuiTableCell-root"
            scope="col"
          >
            Brand
          </th>
          <th
            aria-sort={null}
            className="MuiTableCell-root MuiTableCell-head MuiTableCell-stickyHeader MuiTableCell-sizeMedium css-5xj66b-MuiTableCell-root"
            scope="col"
          >
            Name
          </th>
        </tr>
      </thead>
      <tbody
        className="MuiTableBody-root css-apqrd9-MuiTableBody-root"
        role={null}
      >
        <tr
          className="MuiTableRow-root css-115oldd-MuiTableRow-root"
          role={null}
        >
          <td
            aria-sort={null}
            className="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1fw17s9-MuiTableCell-root"
          />
          <td
            aria-sort={null}
            className="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1fw17s9-MuiTableCell-root"
          />
          <td
            aria-sort={null}
            className="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1fw17s9-MuiTableCell-root"
          />
        </tr>
        <tr
          className="MuiTableRow-root css-115oldd-MuiTableRow-root"
          role={null}
        >
          <td
            aria-sort={null}
            className="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1fw17s9-MuiTableCell-root"
          />
          <td
            aria-sort={null}
            className="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1fw17s9-MuiTableCell-root"
          />
          <td
            aria-sort={null}
            className="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1fw17s9-MuiTableCell-root"
          />
        </tr>
      </tbody>
    </table>
  </div>,
  <div
    className="MuiBox-root css-1w1vbxm"
  >
    <button
      className="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorPrimary MuiIconButton-sizeMedium css-1d6pat7-MuiButtonBase-root-MuiIconButton-root"
      disabled={false}
      onBlur={[Function]}
      onClick={[Function]}
      onContextMenu={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={0}
      type="submit"
      variant="outlined"
    >
      <svg
        aria-hidden={true}
        className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
        data-testid="DeleteIcon"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M6 19c0 1.1.9 2 2 2h8c1.1 0 2-.9 2-2V7H6v12zM19 4h-3.5l-1-1h-5l-1 1H5v2h14V4z"
        />
      </svg>
    </button>
    <button
      className="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorPrimary MuiIconButton-sizeMedium css-1d6pat7-MuiButtonBase-root-MuiIconButton-root"
      disabled={false}
      onBlur={[Function]}
      onClick={[Function]}
      onContextMenu={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={0}
      type="submit"
      variant="outlined"
    >
      <svg
        aria-hidden={true}
        className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
        data-testid="AddIcon"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M19 13h-6v6h-2v-6H5v-2h6V5h2v6h6v2z"
        />
      </svg>
    </button>
  </div>,
]
`;
